homeassistant/components/generic_thermostat/climate.py

```diff
@@ -259,6 +259,11 @@ def _async_startup(*_):
         if not self._hvac_mode:
             self._hvac_mode = HVAC_MODE_OFF
 
+        # Prevent the device from keep running if HVAC_MODE_OFF
+        if self._hvac_mode == HVAC_MODE_OFF and self._is_device_active:
+            await self._async_heater_turn_off()
+            _LOGGER.warning("turning off heater %s", self.heater_entity_id)
```

Comment:
If a warning is needed here, please make it more descriptive
Is it abnormal to turn off the heater here? Then the warning should explain why.

Reply:
There is a potential risk that during startup (eg: after a reboot) the climate remains in the off state (and does not control the thermostat) but the switch remains on.
Due to this risk, I believe that the behavior should be to disconnect the switch, even if it is not a function of the integration.
That 'strange' behavior is what the log message is meant to reflect.

I check it to make it more descriptive.

Reply:
Does it seem appropriate to you?

Reply:
Yes, it looks good!
If you apply the suggestion from black as well, we're good to go.